From 58b4de30553c25f2d848d624f1771578654dc4b8 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 18 Aug 2018 14:39:33 +0530
Subject: [PATCH 0135/1051] net: octeontx2: Defer probe if AF is not responding

If RVU AF is not responding to mailbox READY message then it's
most likely that AF driver probe is not done yet. So defer
netdev driver probe to check again

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_pf.c | 2 +-
 drivers/net/ethernet/marvell/octeontx2/otx2_vf.c | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
index 0e35223abde9..5ca9bfaa35d0 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
@@ -274,7 +274,7 @@ static int otx2_register_mbox_intr(struct otx2_nic *pf)
 	if (err) {
 		dev_warn(pf->dev,
 			 "AF not responding to mailbox, deferring probe\n");
-		return err;
+		return -EPROBE_DEFER;
 	}
 	return 0;
 }
diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_vf.c
index 268d51b14f2b..d75c71a522a1 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_vf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_vf.c
@@ -181,8 +181,9 @@ static int otx2vf_register_mbox_intr(struct otx2_nic *vf)
 	otx2_mbox_alloc_msg_READY(&vf->mbox);
 	err = otx2_sync_mbox_msg(&vf->mbox);
 	if (err) {
-		dev_err(vf->dev, "AF is not responding to mailbox\n");
-		return err;
+		dev_warn(vf->dev,
+			 "AF not responding to mailbox, deferring probe\n");
+		return -EPROBE_DEFER;
 	}
 
 	return 0;
-- 
2.17.1

